language/diem-framework/modules/0L/Teams.move

```diff
@@ -0,0 +1,326 @@
+      collective_tower_height_this_epoch: u64,
+
+      // Informational fields
+      team_name: vector<u8>, // A validator account.
+      description: vector<u8>,
+      count_all_members: u64, // count the miners who have expressed to me members of this team, but may be below threshold
+      count_active: u64, // the count of players above threshold
+
+      // Is team in validator set
+    }
+
+    // this struct is stored in the member's account
+    struct Member has key, copy, drop, store {
+      captain_address: address, // by address of captain
+      mining_above_threshold: bool, // if the mining the user has done is above the system threshold to count toward delegation.
+
+    }
+
+    public fun vm_init(sender: &signer) {
+      CoreAddresses::assert_vm(sender);
+      if (!exists<AllTeams>(CoreAddresses::VM_RESERVED_ADDRESS())) {
+        move_to<AllTeams>(
+          sender, 
+          AllTeams {
+            teams_list: Vector::empty(),
+            collective_threshold_epoch: 0,
+            member_threshold_epoch: 0,
+            tower_height_rms: 0,
+          }
+        );
+      }
+    }
+
+
+    public fun team_init(sender: &signer, team_name: vector<u8>, operator_pct_reward: u64) {
+      if (operator_pct_reward < 10 || operator_pct_reward > 100 ) {
+        return
+      };
+
+      assert(ValidatorUniverse::is_in_universe(Signer::address_of(sender)), 201301001);
+      // An "captain", who is already a validator account, stores the Team struct on their account.
+      // the AllTeams struct is saved in the 0x0 account, and needs to be initialized before this is called.
+
+      // check vm has initialized the struct, otherwise exit early.
+      if (!exists<AllTeams>(CoreAddresses::VM_RESERVED_ADDRESS())) {
+        return
+    };
+
+    move_to<Team>(
+        sender, 
+        Team {
+          captain: Signer::address_of(sender), // A validator account.
+          members: Vector::empty<address>(),
+          operator_pct_reward, // the percentage of the rewards that the captain proposes to go to the validator operator.
+          collective_tower_height_this_epoch: 0,
+
+          team_name, // A validator account.
+          description: Vector::empty<u8>(), // TODO: Change this
+          count_all_members: 0,
+          count_active: 0,
+
+        }
+      );
+    }
+
+
+    // a member states his team's preference.
+    // the member only actually joins a team for the purposes of calculating rewards
+    // when they do enough mining in a period
+    public fun join_team(sender: &signer, new_captain: address) acquires Member, Team {
+      let addr = Signer::address_of(sender);
+      // needs to check if this is a slow wallet.
+      // ask user to resubmit if not a slow wallet, so they are explicitly setting it, no surprises, no tears.
+
+      assert(DiemAccount::is_slow(addr), ENOT_SLOW_WALLET);
```

Comment:
It might be a good idea to wait at least 1 epoch for a new team be able to receive new members because it creates anticipation for Carpe users, who can see new captains donation plans, and avoids validators to create their teams with few members near by the end of an epoch.